Add in propagation of error for convolution smoothing (#486)
astropy/specutils: docs/manipulation.rst

In this case, the ``spec1_bsmooth2`` result should be equivalent to the ``spec1_bsmooth`` in
the section above (assuming the flux data of the input ``spec`` is the same).

> The errors are propagated using a standrard "propagation of errors" method, if the uncertainty 

Reviewer: (minor)
```suggestion
The uncertainties are propagated using a standard "propagation of errors" method, if the uncertainty 
```

is defined for the spectrum *and* it is one of StdDevUncertainty, VarianceUncertainty or InverseVariance.

> *Note:* this does *not* consider covariance between points.

Reviewer: ```suggestion
But note that this does *not* consider covariance between points.
```

Or if you want an actual note "box" (a la https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html#notes) , this could also be

```suggestion

.. note::
    This does *not* consider covariance between points.
    
```

has a similar call structure to the convolution-based smoothing methods. This
method applys the median filter across the flux.

> Note: This method is not flux conserving and errors are not propagated.

Reviewer: ```suggestion
Note that this method is not flux conserving and errors are not propagated.
```

Or it could be
```suggestion

.. note::
    This method is not flux conserving and errors are not propagated.
    
```

Related question: are they not propoagated because we just haven't implemented it yet, or because it's ambiguous what the right way is to do the propogation?  If it's the latter we should probably say that here (i.e. it's a feature not a bug :wink:)
